Sources/PCRoom.swift

```diff
@@ -84,6 +96,18 @@ public final class PCRoom {
                 (self.customData != nil && room.customData != nil &&
                     (self.customData! as NSDictionary).isEqual(to: room.customData!)
                 )
+            ) &&
+            (
+                (self.unreadCount == nil && room.unreadCount == nil) ||
```

Comment:
this looks fascinating. Why is it not sufficient to check `self.unreadCount == room.unreadCount`?

Reply:
You're right! I thought there might be some quirkiness with checking for equality on Optionals, but according to the implementation they work like they should. ie. The operands are equal if they are both nil, or if they are both not nil and the unwrapped values are equal.

Will change.